spark/src/main/java/org/opensearch/sql/spark/client/EMRServerlessClientFactory.java

```diff
@@ -0,0 +1,12 @@
+package org.opensearch.sql.spark.client;
+
+/** Factory interface for creating instances of {@link EMRServerlessClient}. */
+public interface EMRServerlessClientFactory {
```

Comment:
Don't have much context here, are we expecting various of EMRS client creation?

Reply:
more of when to create a new client and return old client.